tests/manage/z_cluster/cluster_expansion/test_delete_osd_pod.py

```diff
@@ -0,0 +1,79 @@
+import pytest
+import logging
+import time
+
+
+from ocs_ci.framework.testlib import ignore_leftovers, ManageTest, tier4a, bugzilla
+from ocs_ci.framework import config
+from ocs_ci.ocs import constants
+from ocs_ci.ocs.ocp import OCP
+from ocs_ci.ocs.resources import pod as pod_helpers
+from ocs_ci.ocs.resources import storage_cluster
+from ocs_ci.ocs.cluster import get_percent_used_capacity
+from ocs_ci.utility.utils import ceph_health_check
+from tests.disruption_helpers import Disruptions
+
+
+@pytest.mark.parametrize(
+    argnames=["workload_storageutilization_rbd"],
+    argvalues=[
+        pytest.param(
+            *[(0.11, True, 120)],
+            marks=pytest.mark.polarion_id("OCS-1203")
+        ),
+    ],
+    indirect=["workload_storageutilization_rbd"],
+)
+@ignore_leftovers
+@tier4a
+@bugzilla('1847098')
+class TestAddCapacityWithOSDPodDelete(ManageTest):
+    """
+    Test add capacity when one of the osd pods gets deleted
+    in the middle of the process.
+    """
+
+    def test_add_capacity_osd_pod_delete(self, workload_storageutilization_rbd):
+        """
+        Test add capacity when one of the osd pods gets deleted
+        in the middle of the process.
+        """
+        used_percentage = get_percent_used_capacity()
+        logging.info(f"storageutilization is completed. used capacity = {used_percentage}")
+
+        max_osds = 15
+        osd_pods_before = pod_helpers.get_osd_pods()
+        if len(osd_pods_before) >= max_osds:
+            pytest.skip("We have maximum of osd's in the cluster")
+
+        d = Disruptions()
+        d.set_resource('osd')
+
+        osd_size = storage_cluster.get_osd_size()
+        logging.info(f"Adding one new set of OSDs. osd size = {osd_size}")
+        storagedeviceset_count = storage_cluster.add_capacity(osd_size)
+        logging.info("Adding one new set of OSDs was issued without problems")
+
+        # OSD number go down by one and then gradually go up by 1
+        # and finally the OSD number will be storagedeviceset_count*3
+        time_to_wait_before_delete_osd_pod = 20
+        time.sleep(time_to_wait_before_delete_osd_pod)
```

Comment:
is there anything we can sample and wait for instead of waiting 20 seconds?

Reply:
I thought to check whether the first new osd pod is in running state, but the process happens really fast, and there is a chance that all the 3 new osd pods will be in running state before I had the chance to delete an osd pod.

Reply:
so in that case, doesn't 20 seconds sleep mean that you could miss the opportunity to delete one of the OSDs before all reach Running? It's not deterministic